modules/config/main.tf

```diff
@@ -0,0 +1,304 @@
+locals {
+  cloudtrail_name = "${var.name}-config-trail"
+
+  cloudtrail_prefix      = var.cloudtrail_prefix == "" ? "gitlab-runner-config" : var.cloudtrail_prefix
+  cloudtrail_bucket      = length(var.cloudtrail_bucket) > 0 ? data.aws_s3_bucket.cloudtrail[0].bucket : "${var.name}-cloudtrail"
+  cloudtrail_bucket_name = length(var.cloudtrail_bucket) > 0 ? data.aws_s3_bucket.cloudtrail[0].bucket : aws_s3_bucket.cloudtrail[0].bucket
+  cloudtrail_bucket_arn  = "arn:aws:s3:::${local.cloudtrail_bucket}"
+
+  config_key         = var.config_key == "" ? "config.toml" : var.config_key
+  config_bucket      = length(var.config_bucket) > 0 ? data.aws_s3_bucket.config[0].bucket : "${var.name}-config"
+  config_bucket_name = length(var.cloudtrail_bucket) > 0 ? data.aws_s3_bucket.config[0].bucket : aws_s3_bucket.config[0].bucket
+  config_bucket_arn  = "arn:aws:s3:::${local.config_bucket}"
+
+  extra_files_prefix = trim(var.extra_files_prefix == "" ? "/extra-files/" : var.extra_files_prefix, "/")
+
+  config_uri      = "s3://${aws_s3_bucket_object.config.bucket}/${aws_s3_bucket_object.config.key}"
```

Comment:
I understand wanting to use the exact values from the `aws_s3_bucket_object` but sadly terraform does not co-operate nicely here: those values are considered dynamic so for every single config change, we also end up generating noise in the terraform plan for any values that depend on `config_uri` and `extra_files_uri`, notably the SSM policy. Instead, if you apply the below diff, even if _content_ of the config file changes, it doesn't produce any extra noise in the plan to do with SSM policies.

```diff
diff --git a/modules/config/main.tf b/modules/config/main.tf
index 6d8f0ad..a1b03aa 100644
--- a/modules/config/main.tf
+++ b/modules/config/main.tf
@@ -13,8 +13,8 @@ locals {
 
   extra_files_prefix = trim(var.extra_files_prefix == "" ? "/extra-files/" : var.extra_files_prefix, "/")
 
-  config_uri      = "s3://${aws_s3_bucket_object.config.bucket}/${aws_s3_bucket_object.config.key}"
-  extra_files_uri = "s3://${aws_s3_bucket_object.config.bucket}/${local.extra_files_prefix}"
+  config_uri      = "s3://${local.config_bucket_name}/${local.config_key}"
+  extra_files_uri = "s3://${local.config_bucket_name}/${local.extra_files_prefix}"
 
   post_reload_script = length(var.post_reload_script) > 0 ? "echo \"Executing post_reload_config script...\"\n${var.post_reload_script}" : ""
 }
```